index.d.ts

```diff
@@ -31,9 +37,8 @@ declare module 'leaflet' {
             'content'?: L.LeafletEventHandlerFn,
         }
 
-        export class Sidebar extends L.Evented { // extends L.Control { // FIXME
```

Comment:
Is #45 fixed also if we keep `L.Evented` here?

Reply:
#45 will not be fixed if we extend L.Evented - since the problem is caused by extending L.Evented since Sidebar is actually an extension of L.Control and not L.Evented.

You also cannot say that L.Evented is an extension of L.Control since it not defined as such in @types/leaflet - they are both extensions of the baseclass L.Class. As such - there will always be type errors if you try.

I am curious to understand what you want to acheive by extending L.Evented? My PR seems to provide all of the methods and properties required. What is missing?

Reply:
As far as I understand it : 

You can created the mixin by adding the following line below the class definition (which in this case would then not extend anything :
```
interface Sidebar extends L.Control, L.Evented {}
```
But this will mean that the class definition for L.Evented.on will clash with the Control.Sidebar.on definition (i.e. #45). You cannot overload the definition for `.on()` with additional types in TS.

You could do the above and delete the defintions for `Control.Sidebar.on()`. This will prevent the type clash and you should then be able to call `Control.Sidebar(type:string)` since the typedef for L.Control.Evented includes :
```
on(type: string, fn: LeafletEventHandlerFn, context?: any): this;
```
But that would mean that the type for Sidebar events must be string - which is not very TS-ish!

If you leave it as per my PR - there is no type clash since we are extending L.Control which has no definition for `Control.on()`. If there are additional events that yoy want to include in the defintion from the L.Evented mixin - just add them to the definition of SidebarEvents etc. It cannot be that many?

Reply:
ok, thanks

>  You cannot overload the definition for .on() with additional types in TS.

That answers my question ;)

I wanted to extend L.Evented, as that is what this the runtime code does, (its just that the @types/leaflet typedef is too strict). But functionally the `extend` doesnt matter, so lets leave it as is.